src/controllers/genresController.ts

```diff
@@ -0,0 +1,71 @@
+import { Request, Response } from 'express';
+import Genre, { IGenreDocument } from '../models/genreModel';
+
+// Get all genres
+export const getGenres = async (req: Request, res: Response): Promise<void> => {
+  try {
+    const genres: IGenreDocument[] = await Genre.find();
+    res.json(genres);
+  } catch (err) {
+    res.status(500).json({ error: 'An error occurred while fetching genres.' });
+  }
+};
+
+// Create a new genre
+export const createGenre = async (req: Request, res: Response): Promise<void> => {
+  try {
+    const { name } = req.body;
+    if (!name) {
+      res.status(400).json({ error: 'Genre name is required.' });
+    }
+
+    const existingGenre = await Genre.findOne({ name });
+    if (existingGenre) {
+      res.status(400).json({ error: 'Genre with the same name already exists.' });
```

Comment:
You have a lot of places with similar response.  Try to Do Not Repeat Yourself. Maybe thwow an error and handle it in error handling middleware.